agents/wiregrid_kikusui/kikusui_agent.py

```diff
@@ -0,0 +1,603 @@
+            if not acquired:
+                self.log.warn(
+                    'Could not run set_c() because {} is already running'
+                    .format(self.lock.job))
+                return False, 'Could not acquire lock'
+
+            if params['current'] <= 3. and 0. <= params['current']:
+                current = params['current']
+                self.cmd.user_input('C {}'.format(params['current']))
+            else:
+                current = 3.0
+                self.log.warn(
+                    'Value Error: set current 3.0 A or less. '
+                    'Now set to {} A'.format(current))
+                self.cmd.user_input('C {}'.format(current))
+
+            return True, 'Set Kikusui current to {} A'.format(current)
+
+    def set_v(self, session, params=None):
+        """
+        Set voltage [V]
+
+        Parameters:
+            volt: set voltage [V] (should be ONLY 12)
+        """
+        if params is None:
+            params = {'volt': 0}
+
+        with self.lock.acquire_timeout(timeout=5, job='set_v') as acquired:
+            if not acquired:
+                self.log.warn(
+                    'Could not run set_v() because {} is already running'
+                    .format(self.lock.job))
+                return False, 'Could not acquire lock'
+
+            if params['volt'] == 12.:
+                self.cmd.user_input('V {}'.format(params['volt']))
+            else:
+                self.log.warn(
+                    'Value Error: Rated Voltage of the motor is 12 V. '
+                    'Now set to 12 V')
+                self.cmd.user_input('V {}'.format(12.))
+
+            return True, 'Set Kikusui voltage to 12 V'
+
+    def get_vc(self, session, params=None):
+        """
+        Show voltage [V], current [A], output on/off
+
+        Parameters:
+            Nothing
+        """
+        with self.lock.acquire_timeout(timeout=5, job='get_vc') as acquired:
+            if not acquired:
+                self.log.warn(
+                    'Could not run get_vc() because {} is already running'
+                    .format(self.lock.job))
+                return False, 'Could not acquire lock'
+
+            v_val = None
+            c_val = None
+            s_val = None
+            msg = 'Error'
+            s_msg = 'Error'
+
+            # check connection
+            ret, msg = self._check_connect()
+            if not ret:
+                msg = 'Could not get c,v because of failure of connection.'
+            else:
+                msg, v_val, c_val = self.cmd.user_input('VC?')
+                s_msg, s_val = self.cmd.user_input('O?')
+
+            self.log.info('Get voltage/current message: {}'.format(msg))
+            self.log.info('Get status message: {}'.format(s_msg))
+            return True,\
+                'Get Kikusui voltage / current: {} V / {} A [status={}]'\
+                .format(v_val, c_val, s_val)
```

Comment:
This isn't the best place to pass that information. If you need the data available to an OCS Client, put it in `session.data`. If you need to record the values to disk for archiving (to .g3 file) publish them to a feed.

Reply:
I removed the two self.log.info() printing lines. Is it what you want?
Here, I want just to show the volt, current, and status. (I don't need to record them.)
I think the return has a message about them. It is enough for me.

Reply:
That's fine if they don't need to be recorded. I just wanted to make sure this didn't need to be acted on programmatically somewhere.